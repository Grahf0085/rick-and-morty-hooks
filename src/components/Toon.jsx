import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Toon = ({ id, name, image }) => (
  <figure>
    <Link to={`/character/${id}`}>
      <img src={image} alt={name} />
    </Link>
  </figure>
);

Toon.propTypes = {
  id:PropTypes.string.isRequired,
  name: PropTypes.string,
  image: PropTypes.string.isRequired
};

export default Toon;
