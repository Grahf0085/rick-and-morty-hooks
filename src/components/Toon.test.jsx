import React from 'react';
import { render } from '@testing-library/react';
import Toon from './Toon';
import { MemoryRouter as Router } from 'react-router-dom';

describe('Toon component', () => {
  it('renders clickable image', () => {
    const url = 'https://rickandmortyapi.com/api/character/avatar/1.jpeg';
    const name = 'Rick Sanchez';
    const id = '1';

    const { asFragment } = render(
      <Router>
        <Toon id={id} image={url} name={name} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
