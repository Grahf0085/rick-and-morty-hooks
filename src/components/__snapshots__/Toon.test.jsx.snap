// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toon component renders clickable image 1`] = `
<DocumentFragment>
  <figure>
    <a
      href="/character/1"
    >
      <img
        alt="Rick Sanchez"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
    </a>
  </figure>
</DocumentFragment>
`;
